src/x.h

```diff
@@ -22,6 +22,11 @@ typedef struct winprop winprop_t;
   r; \
 })
 
+static inline void
+x_sync(xcb_connection_t *c) {
```

Comment:
Can you document why `xcb_get_input_focus` is used for sync here? You can simply point to some document online.

Reply:
Well... I do not really know what to point at.

There is the implementation of `XSync`: https://sources.debian.org/src/libx11/2:1.6.6-1/src/Sync.c/?hl=35#L43

Then there is libxcb-util with header `xcb_aux.h` which provides a function `xcb_aux_sync` that does just the same thing.

I guess this is just "the X11 server processes requests in-order, so it can only reply to GetInputFocus after having handling all previous requests".
(However, with Xlib, the library will also already have handled any errors that occurred while XCB just queues them for later `xcb_wait_for_event`...)

Reply:
OK, I will put something there.